internal/unit-info-viewer/skills-progress.lua

```diff
@@ -0,0 +1,181 @@
+--@ module=true
+
+local utils = require("utils")
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+local view_sheets = df.global.game.main_interface.view_sheets
+
+local function get_skill(id)
+    return utils.binsearch(
+        df.unit.find(view_sheets.viewing_unid[0]).status.current_soul.skills,
+        view_sheets.unit_skill[id],
+        "id"
+    )
+end
+
+SkillProgressOverlay=defclass(SkillProgressOverlay, overlay.OverlayWidget)
+SkillProgressOverlay.ATTRS {
+    desc="Display progress bars for learning skills!",
+    default_pos={x=-43,y=20},
+    default_enabled=true,
+    viewscreens= {
+        'dwarfmode/ViewSheets/UNIT/Skills/Labor',
+        'dwarfmode/ViewSheets/UNIT/Skills/Combat',
+        'dwarfmode/ViewSheets/UNIT/Skills/Social',
+        'dwarfmode/ViewSheets/UNIT/Skills/Other',
+
+        'dungeonmode/ViewSheets/UNIT/Skills/Labor',
+        'dungeonmode/ViewSheets/UNIT/Skills/Combat',
+        'dungeonmode/ViewSheets/UNIT/Skills/Social',
+        'dungeonmode/ViewSheets/UNIT/Skills/Other',
+    },
+    frame={w=54, h=67},
+}
+
+function SkillProgressOverlay:init()
+    self.progress_bar = true
+    self.display_experience = true
+
+    self:addviews{
+        widgets.Label{
+            view_id='annotations',
+            frame={t=0, l=self.frame.w-16, w=16},
+            text='',
+            text_pen=COLOR_GRAY,
+        },
+        widgets.ToggleHotkeyLabel{
+            frame={t=self.frame.h-1,l=2, w=25},
+            label='Progress Bar',
+            key='CUSTOM_CTRL_B',
+            options={
+                {label='No', value=false, pen=COLOR_WHITE},
+                {label='Yes', value=true, pen=COLOR_YELLOW},
+            },
+            view_id='toggle_progress',
+            on_change=function(option) self.progress_bar = option end,
+        },
+        widgets.ToggleHotkeyLabel{
+            frame={t=self.frame.h-1,l=28, w=25},
+            label='Experience  ',
+            key='CUSTOM_CTRL_E',
+            options={
+                {label='No', value=false, pen=COLOR_WHITE},
+                {label='Yes', value=true, pen=COLOR_YELLOW},
+            },
+            view_id='toggle_experience',
+            on_change=function(option) self.display_experience = option end,
+        },
+    }
+end
+
+-- Use render to set On/Off dynamically for each unit
+function SkillProgressOverlay:render(dc)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if unit then
+        self.subviews.toggle_progress:setOption(self.progress_bar)
+        self.subviews.toggle_experience:setOption(self.display_experience)
+    end
+
+    SkillProgressOverlay.super.render(self, dc)
+end
```

Comment:
I'm not sure I understand what this is doing. why would the options need to be restored?